### PR TITLE
Feature external constraints metaschema models

### PR DESCRIPTION
**Pull request by david-waltermire**
# Committer Notes

Model legacy and new external constraints formats as a Metaschema module.

### All Submissions:

- [ ] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.

### CODE REVIEW

**Comment by wendellpiez** [comment]
Awesome to see progress - let's see some examples! 🥬

File: schema/metaschema/metaschema-module-constraints.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<metaschema-meta-constraints xmlns="http://csrc.nist.gov/ns/oscal/metaschema/1.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://csrc.nist.gov/ns/oscal/metaschema/1.0 file:/C:/Users/davidwal/git/github/david-waltermire-nist/metaschema/metaschema-java/core/metaschema/schema/xml/metaschema-meta-constraints.xsd">
```
Comment (aj-stein-nist) [inline schema/metaschema/metaschema-module-constraints.xml:4]: Minor path issues with schemLocation binding.

```suggestion
    xsi:schemaLocation="http://csrc.nist.gov/ns/oscal/metaschema/1.0 ../xml/metaschema-meta-constraints.xsd">
```

File: schema/metaschema/metaschema-module-metaschema.xml [hunk 2]
```diff
@@ -1187,4 +1141,195 @@
...
+                <group-as name="metapath-contexts" in-json="ARRAY"/>
+            </assembly>
+        </model>            
+    </define-assembly>
+
+	<define-assembly name="metaschema-metapath">
+	    <description>A Metapath expression identifying the model node that the constraints will be applied to.</description>
+	    <use-name>metapath</use-name>
+	    <define-flag name="target" required="yes"/>
+	</define-assembly>
```
Comment (aj-stein-nist) [inline schema/metaschema/metaschema-module-metaschema.xml:1318]:
```suggestion
   <define-assembly name="metaschema-metapath">
      <description>A Metapath expression identifying the model node that the constraints will be applied to.</description>
      <use-name>metapath</use-name>
      <define-flag name="target" required="yes"/>
   </define-assembly>
```

**Review by aj-stein-nist — APPROVED** [review]
Minor nits we discussed while pairing, looks good! I will test and build test/WIP models with it now. 😄

**Review by aj-stein-nist — APPROVED** [review]

**Review by aj-stein-nist — APPROVED** [review]

**Pull request merged**